app/controllers/merchant_invoices_controller.rb

```diff
@@ -2,4 +2,9 @@ class MerchantInvoicesController < ApplicationController
   def index
     @merchant = Merchant.find(params[:merchant_id])
   end
+
+  def show 
+    @invoice = Invoice.find(params[:invoice_id])
+    @customer = Customer.find(@invoice.customer_id)
```

Comment:
I think instead of making a 2nd instance variable for customer, you can call @invoice.customer in the view if you need the customer?  I hadn't seen @customer used in the view, so we may not even need this?  Just trying to think about keeping the controllers skinny.